R/SummaryPanel4a.R

```diff
@@ -0,0 +1,67 @@
+#' Module UI for the covariate summary plot tab.
+#' 
+#' @param id ID of the module
+#' @return Tab 4a Summary
+metaregression_summary_panel_ui <- function(id) {
+  ns <- NS(id)
+  fluidPage(
+    h4('Summary Characteristic Plot'),
+    uiOutput(ns('toggle_covariate_baseline')),
```

Comment:
Can you make this a `shinyWidgets::radioGroupButtons` then use `shinyWidgets::updateRadioGroupButtons()` in the server please? It's best to have as few `uiOutput`s as possible because it pollutes the server logic.

Reply:
The momentary error is down to this being rendered later, so the value is not available when the plot is made.

The best fix would be to wrap the `shinyWidgets::updateRadioGroupButtons()` in an `observe` with a priority of 1 to make sure that the input is available before the plot is rendered.

As an aside, it's best to give all inputs and outputs unique names to avoid confusion.

Reply:
> The momentary error is down to this being rendered later, so the value is not available when the plot is made.
> 
> The best fix would be to wrap the `shinyWidgets::updateRadioGroupButtons()` in an `observe` with a priority of 1 to make sure that the input is available before the plot is rendered.

I tried this and am still getting the error.

Reply:
> As an aside, it's best to give all inputs and outputs unique names to avoid confusion.

Changed

Reply:
I managed to fix the error by swapping the `input$toggle_covariate_baseline` to `ifelse(is.null(input$toggle_covariate_baseline), 'Baseline risk', input$toggle_covariate_baseline)` in the call to `CreateCovariateSummaryPlot()`.

This is also a bit cleaner as the check in the plot function: `covariate_or_baseline == "Baseline risk"` evaluates to an empty vector `logical(0)` when `covariate_or_baseline` is `NULL`.

Reply:
> This is also a bit cleaner as the check in the plot function: `covariate_or_baseline == "Baseline risk"` evaluates to an empty vector `logical(0)` when `covariate_or_baseline` is `NULL`.

I never liked that line so this is much better.